include/cuco/detail/static_multimap/device_view_impl.inl

```diff
@@ -502,19 +503,23 @@ class static_multimap<Key, Value, Scope, Allocator, ProbeSequence>::device_view_
 #if defined(CUCO_HAS_CUDA_BARRIER)
       cooperative_groups::memcpy_async(
         g,
-        output_begin + offset,
+        &thrust::raw_reference_cast(*(output_begin + offset)),
         output_buffer,
         cuda::aligned_size_t<alignof(value_type)>(sizeof(value_type) * num_outputs));
 #else
-      cooperative_groups::memcpy_async(
-        g, output_begin + offset, output_buffer, sizeof(value_type) * num_outputs);
+      cooperative_groups::memcpy_async(g,
+                                       &thrust::raw_reference_cast(*(output_begin + offset)),
+                                       output_buffer,
+                                       sizeof(value_type) * num_outputs);
 #endif  // end CUCO_HAS_CUDA_BARRIER
       return;
 #endif  // end CUCO_HAS_CG_MEMCPY_ASYNC
     }
+#pragma nv_diag_suppress 128  // warning: unreachable
     for (auto index = lane_id; index < num_outputs; index += g.size()) {
       *(output_begin + offset + index) = output_buffer[index];
     }
+#pragma nv_diag_default 128
```

Comment:
I thought I had fixed the following warning with the `nv_diag_suppress`:
```
include/cuco/detail/static_multimap/device_view_impl.inl(519): warning: loop is not reachable
```
but it still pops up.

I'll see if I can reorder the logic in order to fix that.